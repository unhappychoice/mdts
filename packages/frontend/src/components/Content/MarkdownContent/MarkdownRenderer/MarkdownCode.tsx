import React from 'react';
import { useSelector } from 'react-redux';
import { PrismAsyncLight } from 'react-syntax-highlighter';
import { SYNTAX_HIGHLIGHTER_THEMES } from '../../../../constants';
import { useSyntaxHighlighterTheme } from '../../../../hooks/useSyntaxHighlighterTheme';
import { RootState } from '../../../../store/store';
import MermaidRenderer from './MermaidRenderer';

interface MarkdownCodeProps {
  inline?: boolean;
  className?: string;
  children?: React.ReactNode;
}

const MarkdownCode: React.FC<MarkdownCodeProps> = ({ inline, className, children, ...props }) => {
  const { fontFamilyMonospace, syntaxHighlighterTheme } = useSelector((state: RootState) => state.config);
  const syntaxHighlighterStyle = useSyntaxHighlighterTheme(syntaxHighlighterTheme);

  const match = /language-(\w+)/.exec(className || '');
  if (match && match[1] === 'mermaid') {
    return <MermaidRenderer chart={String(children).replace(/\n$/, '')} />;
  }

  return !inline && match ? (
    <PrismAsyncLight
      style={syntaxHighlighterStyle}
      className={'syntax-highlighter'}
      customStyle={{ margin: 0 }}
      showLineNumbers={true}
      language={match[1]}
      codeTagProps={{ style: { fontFamily: fontFamilyMonospace } }}
      PreTag="div"
      {...props}
    >
      {String(children).replace(/\n$/, '')}
    </PrismAsyncLight>
  ) : (
    <code
      {...props}
      className={className}
      style={{
        color: SYNTAX_HIGHLIGHTER_THEMES.find(theme => theme.value === syntaxHighlighterTheme)?.type === 'light'
          ? 'rgba(0, 0, 0, .87)'
          : '#fff',
        fontFamily: fontFamilyMonospace,
        backgroundColor: SYNTAX_HIGHLIGHTER_THEMES.find(theme => theme.value === syntaxHighlighterTheme)?.color,
        borderRadius: '4px',
      }}
    >
      {children}
    </code>
  );
};

export default MarkdownCode;
