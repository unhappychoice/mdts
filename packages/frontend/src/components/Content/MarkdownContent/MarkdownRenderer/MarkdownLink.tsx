import React, { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';

interface MarkdownLinkProps {
  href?: string;
  children?: React.ReactNode;
  selectedFilePath: string | null;
}

const MarkdownLink: React.FC<MarkdownLinkProps> = ({ href, children, selectedFilePath }) => {
  const navigate = useNavigate();

  const resolvePath = useCallback((href: string) => {
    if (!selectedFilePath || href.startsWith('http') || href.startsWith('//') || href.startsWith('/')) {
      return href;
    }
    const baseUrl = selectedFilePath.substring(0, selectedFilePath.lastIndexOf('/') + 1);
    return new URL(href, new URL(baseUrl, window.location.origin)).pathname;
  }, [selectedFilePath]);

  const handleClick = useCallback((e: React.MouseEvent<HTMLAnchorElement>) => {
    if (!href) return;
    const resolvedHref = resolvePath(href);
    e.preventDefault();
    navigate(resolvedHref);
  }, [href, resolvePath, navigate]);

  if (!href) {
    return <a data-testid="markdown-link">{children}</a>;
  }
  if (href.startsWith('#')) {
    return <a data-testid="markdown-link" href={href}>{children}</a>;
  }

  const resolvedHref = resolvePath(href);
  if (resolvedHref.startsWith('http') || resolvedHref.startsWith('//')) {
    return <a data-testid="markdown-link" href={resolvedHref} target="_blank" rel="noopener noreferrer">{children}</a>;
  }

  return <a data-testid="markdown-link" href={resolvedHref} onClick={handleClick}>{children}</a>;
};

export default MarkdownLink;
