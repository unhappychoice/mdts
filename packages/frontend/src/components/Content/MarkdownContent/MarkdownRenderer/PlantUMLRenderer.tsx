import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { RootState, AppDispatch } from '../../../../store/store';
import { generatePlantUMLSvg, selectPlantUMLData } from '../../../../store/slices/plantUMLSlice';

interface PlantUMLProps {
  chart: string;
}


const PlantUMLRenderer: React.FC<PlantUMLProps> = ({ chart }) => {
  const dispatch = useDispatch<AppDispatch>();
  const { svg, isLoading, error } = useSelector((state: RootState) => 
    selectPlantUMLData(state, chart)
  );

  useEffect(() => {
    if (chart && !svg && !isLoading) {
      dispatch(generatePlantUMLSvg(chart));
    }
  }, [chart, svg, isLoading, dispatch]);

  if (error) {
    return (
      <div style={{ color: 'red', padding: '8px', border: '1px solid red', borderRadius: '4px' }}>
        <strong>PlantUML Error:</strong> {error}
        <pre><code>{chart}</code></pre>
      </div>
    );
  }

  if (isLoading) {
    return <div>Loading PlantUML diagram...</div>;
  }

  if (!svg) {
    return <pre><code>{chart}</code></pre>;
  }

  return <div dangerouslySetInnerHTML={{ __html: svg }} style={{ maxWidth: '100%' }} />;
};

export default PlantUMLRenderer;
