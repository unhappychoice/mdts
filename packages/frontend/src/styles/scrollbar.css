.custom-scrollbar {
  overflow: auto;
}

.custom-scrollbar::-webkit-scrollbar {
  width: 8px;
}

.custom-scrollbar::-webkit-scrollbar-track {
  background: transparent;
}

.custom-scrollbar::-webkit-scrollbar-thumb {
  background: transparent; /* Initially transparent */
  border-radius: 4px;
}

/* Light mode: Show scrollbar thumb on hover */
.custom-scrollbar:hover::-webkit-scrollbar-thumb {
  background: var(--scrollbar-thumb-color);
}

/* Light mode: Scrollbar thumb hover color */
.custom-scrollbar::-webkit-scrollbar-thumb:hover {
  background: var(--scrollbar-thumb-hover-color);
}

/* Dark mode: Show scrollbar thumb on hover */
.custom-scrollbar:hover::-webkit-scrollbar-thumb {
  background: var(--scrollbar-thumb-color);
}

.custom-scrollbar::-webkit-scrollbar-thumb:hover {
  background: var(--scrollbar-thumb-hover-color);
}