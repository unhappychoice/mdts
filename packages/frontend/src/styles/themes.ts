interface ThemePalette {
  primary: string;
  background: string;
  paper: string;
}

export interface AppTheme {
  id: string;
  name: string;
  light: ThemePalette;
  dark: ThemePalette;
}

export const themes: AppTheme[] = [
  {
    id: 'default',
    name: 'Default',
    light: {
      primary: '#1976d2',
      background: '#f5f7fa',
      paper: '#eef1f5',
    },
    dark: {
      primary: '#1976d2',
      background: '#161819',
      paper: '#0f1214',
    },
  },
  {
    id: 'aurora',
    name: 'Aurora',
    light: {
      primary: '#00796b',
      background: '#e0f2f1',
      paper: '#d0eceb',
    },
    dark: {
      primary: '#80cbc4',
      background: '#00251a',
      paper: '#012a4a',
    },
  },
  {
    id: 'autumn',
    name: 'Autumn',
    light: {
      primary: '#bf360c',
      background: '#fff3e0',
      paper: '#ffe0b2',
    },
    dark: {
      primary: '#ff8a65',
      background: '#3e2723',
      paper: '#2a1a1a',
    },
  },
  {
    id: 'cafe',
    name: 'Caf√© Mocha',
    light: {
      primary: '#795548',
      background: '#f9f3ef',
      paper: '#f3ebe5',
    },
    dark: {
      primary: '#a1887f',
      background: '#2e1f1a',
      paper: '#1e1510',
    },
  },
  {
    id: 'deepsea',
    name: 'Deep Sea',
    light: {
      primary: '#01579b',
      background: '#e1f1fa',
      paper: '#d4ebf6',
    },
    dark: {
      primary: '#0288d1',
      background: '#0b1e2b',
      paper: '#10222f',
    },
  },
  {
    id: 'forest',
    name: 'Forest',
    light: {
      primary: '#2e7d32',
      background: '#eef5f0',
      paper: '#e6f2eb',
    },
    dark: {
      primary: '#4caf50',
      background: '#1a2a1a',
      paper: '#111f11',
    },
  },
  {
    id: 'glacier',
    name: 'Glacier',
    light: {
      primary: '#00acc1',
      background: '#e0f7fa',
      paper: '#d1eef2',
    },
    dark: {
      primary: '#4dd0e1',
      background: '#0d2b2f',
      paper: '#112a2d',
    },
  },
  {
    id: 'inkblue',
    name: 'Ink Blue',
    light: {
      primary: '#3f51b5',
      background: '#e8eaf6',
      paper: '#dee1f3',
    },
    dark: {
      primary: '#7986cb',
      background: '#1a1c2b',
      paper: '#121522',
    },
  },
  {
    id: 'inkstone',
    name: 'Inkstone',
    light: {
      primary: '#37474f',
      background: '#f3f6f8',
      paper: '#e6ebee',
    },
    dark: {
      primary: '#90a4ae',
      background: '#1c1f22',
      paper: '#121518',
    },
  },
  {
    id: 'lavender',
    name: 'Lavender',
    light: {
      primary: '#9c27b0',
      background: '#f5ecfb',
      paper: '#f1e3f8',
    },
    dark: {
      primary: '#ba68c8',
      background: '#2a1a2f',
      paper: '#1c1023',
    },
  },
  {
    id: 'moss',
    name: 'Moss',
    light: {
      primary: '#558b2f',
      background: '#eef5e4',
      paper: '#e2efdc',
    },
    dark: {
      primary: '#9ccc65',
      background: '#1a2d1a',
      paper: '#152215',
    },
  },
  {
    id: 'nightfox',
    name: 'Nightfox',
    light: {
      primary: '#546e7a',
      background: '#f0f4f8',
      paper: '#e6edf2',
    },
    dark: {
      primary: '#8faac2',
      background: '#0d1117',
      paper: '#161b22',
    },
  },
  {
    id: 'ocean',
    name: 'Ocean',
    light: {
      primary: '#00838f',
      background: '#e0f4f7',
      paper: '#d6f0f2',
    },
    dark: {
      primary: '#00bcd4',
      background: '#1a2a2d',
      paper: '#111f22',
    },
  },
  {
    id: 'rosequartz',
    name: 'Rose Quartz',
    light: {
      primary: '#e91e63',
      background: '#fce4ec',
      paper: '#f9dce7',
    },
    dark: {
      primary: '#f06292',
      background: '#2c1823',
      paper: '#1a0e15',
    },
  },
  {
    id: 'sakura',
    name: 'Sakura',
    light: {
      primary: '#e65a67',
      background: '#fef1f2',
      paper: '#ffeef0',
    },
    dark: {
      primary: '#e65a67',
      background: '#2a1a1f',
      paper: '#1f1115',
    },
  },
  {
    id: 'sandstone',
    name: 'Sandstone',
    light: {
      primary: '#c77966',
      background: '#fdf4ec',
      paper: '#f9eae0',
    },
    dark: {
      primary: '#e57373',
      background: '#3e2a25',
      paper: '#291b17',
    },
  },
  {
    id: 'solarized',
    name: 'Solarized',
    light: {
      primary: '#268bd2',
      background: '#fdf6e3',
      paper: '#f4efdc',
    },
    dark: {
      primary: '#b58900',
      background: '#002b36',
      paper: '#073642',
    },
  },
  {
    id: 'storm',
    name: 'Storm',
    light: {
      primary: '#455a64',
      background: '#e9edf0',
      paper: '#dde3e7',
    },
    dark: {
      primary: '#90a4ae',
      background: '#1e262b',
      paper: '#13191e',
    },
  },
  {
    id: 'sunset',
    name: 'Sunset',
    light: {
      primary: '#d84315',
      background: '#fff1e6',
      paper: '#fde8dd',
    },
    dark: {
      primary: '#ff8a65',
      background: '#2c1e1a',
      paper: '#2a1a2f',
    },
  },
  {
    id: 'tropical',
    name: 'Tropical',
    light: {
      primary: '#ef6c00',
      background: '#fff9e5',
      paper: '#fff0dd',
    },
    dark: {
      primary: '#ffb74d',
      background: '#3e2723',
      paper: '#00363a',
    },
  },
  {
    id: 'wisteria',
    name: 'Wisteria',
    light: {
      primary: '#7e57c2',
      background: '#f3e5f5',
      paper: '#e9dcf0',
    },
    dark: {
      primary: '#9575cd',
      background: '#291b3a',
      paper: '#1b1027',
    },
  },
];
