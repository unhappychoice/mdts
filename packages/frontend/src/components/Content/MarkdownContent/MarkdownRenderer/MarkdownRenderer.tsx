import { Box } from '@mui/material';
import React from 'react';
import ReactMarkdown from 'react-markdown';
import rehypeRaw from 'rehype-raw';
import remarkGfm from 'remark-gfm';
import remarkSlug from 'remark-slug';
import MarkdownLink from './MarkdownLink';
import MarkdownCode from './MarkdownCode';

interface MarkdownRendererProps {
  content: string;
  selectedFilePath: string | null;
}

const MarkdownRenderer: React.FC<MarkdownRendererProps> = ({ content, selectedFilePath }) => {
  return (
    <Box className={'markdown-body'} sx={{ py: 2, px: 0 }}>
      <ReactMarkdown
        remarkPlugins={[remarkGfm, remarkSlug]}
        rehypePlugins={[rehypeRaw]}
        components={{
          a: ({ href, children }) =>
            <MarkdownLink href={href} selectedFilePath={selectedFilePath}>{children}</MarkdownLink>,
          code: ({ inline, className, children, ...props }) => (
            <MarkdownCode inline={inline} className={className} {...props}>
              {children}
            </MarkdownCode>
          ),
        }}
      >
        {content}
      </ReactMarkdown>
    </Box>
  );
};

export default MarkdownRenderer;
