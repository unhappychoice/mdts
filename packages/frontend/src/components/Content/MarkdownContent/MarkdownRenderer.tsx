import { Box, useTheme } from '@mui/material';
import React from 'react';
import { useNavigate } from 'react-router-dom';
import ReactMarkdown from 'react-markdown';
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { nightOwl, prism } from 'react-syntax-highlighter/dist/esm/styles/prism';
import rehypeRaw from 'rehype-raw';
import remarkGfm from 'remark-gfm';
import remarkSlug from 'remark-slug';
import MermaidRenderer from './MermaidRenderer';

interface MarkdownRendererProps {
  content: string;
  selectedFilePath: string | null;
}

const MarkdownRenderer: React.FC<MarkdownRendererProps> = ({ content, selectedFilePath }) => {
  const theme = useTheme();
  const navigate = useNavigate();

  const resolvePath = (href: string) => {
    if (!selectedFilePath || href.startsWith('http') || href.startsWith('//') || href.startsWith('/')) {
      return href;
    }
    const baseUrl = selectedFilePath.substring(0, selectedFilePath.lastIndexOf('/') + 1);
    return new URL(href, new URL(baseUrl, window.location.origin)).pathname;
  };

  return (
    <Box className={"markdown-body"} sx={{ py: 2, px: 0, fontSize: '0.9rem' }}>
      <ReactMarkdown
        remarkPlugins={[remarkGfm, remarkSlug]}
        rehypePlugins={[rehypeRaw]}
        components={{
          a: ({ href, children }) => {
            if (!href) {
              return <a data-testid="markdown-link">{children}</a>;
            }
            if (href.startsWith('#')) {
              return <a data-testid="markdown-link" href={href}>{children}</a>;
            }
            const resolvedHref = resolvePath(href);
            if (resolvedHref.startsWith('http') || resolvedHref.startsWith('//')) {
              return <a data-testid="markdown-link" href={resolvedHref} target="_blank" rel="noopener noreferrer">{children}</a>;
            }
            return <a data-testid="markdown-link" href={resolvedHref} onClick={(e) => { e.preventDefault(); navigate(resolvedHref); }}>{children}</a>;
          },
          code({ node, inline, className, children, ...props }) {
            const match = /language-(\w+)/.exec(className || '');
            if (match && match[1] === 'mermaid') {
              return <MermaidRenderer chart={String(children).replace(/\n$/, '')} />;
            }

            return !inline && match ? (
              <SyntaxHighlighter
                style={theme.palette.mode === 'dark' ? nightOwl : prism}
                className={'syntax-highlighter'}
                customStyle={{  margin: 0, background: 'transparent' }}
                showLineNumbers={true}
                language={match[1]}
                PreTag="div"
                {...props}
              >
                {String(children).replace(/\n$/, '')}
              </SyntaxHighlighter>
            ) : (
              <code className={className} {...props}>
                {children}
              </code>
            );
          },
        }}
      >
        {content}
      </ReactMarkdown>
    </Box>
  );
};

export default MarkdownRenderer;
