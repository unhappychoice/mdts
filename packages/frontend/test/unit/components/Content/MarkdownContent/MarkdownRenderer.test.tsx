import { createTheme, ThemeProvider } from '@mui/material/styles';
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import { BrowserRouter, useNavigate } from 'react-router-dom';
import MarkdownRenderer from '../../../../../src/components/Content/MarkdownContent/MarkdownRenderer';

// Mock react-router-dom's useNavigate
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

// Mock Mermaid component
jest.mock('../../../../../src/components/Content/MarkdownContent/MermaidRenderer', () => ({
  __esModule: true,
  default: ({ chart }: { chart: string }) => <div data-testid="mock-mermaid-chart">{chart}</div>,
}));

// Mock ReactMarkdown to pass through children and allow inspection of components prop
jest.mock('react-markdown', () => {
  const React = jest.requireActual('react');
  const ActualReactMarkdown = jest.requireActual('react-markdown').default;

  
  const MockCode = (props: any) => {
    if (props.className && props.className.includes('language-mermaid')) {
      return <div data-testid="mock-mermaid-chart">{props.children}</div>;
    }
    return (
      <pre data-testid="mock-syntax-highlighter" className={props.className}>
        <code data-testid="mock-syntax-highlighter-code">{props.children}</code>
      </pre>
    );
  };
  const MockH1 = (props: any) => <h1 data-testid="mock-h1" {...props}>{props.children}</h1>;
  const MockP = (props: any) => <p data-testid="mock-p" {...props}>{props.children}</p>;

  return {
    __esModule: true,
    default: ({ children, components, remarkPlugins, rehypePlugins }: { children: string; components?: any; remarkPlugins?: any[]; rehypePlugins?: any[] }) => {
      const finalComponents = {
        ...components,
        
        code: MockCode,
        h1: MockH1,
        p: MockP,
      };
      return (
        <div data-testid="mock-react-markdown-root">
          <ActualReactMarkdown components={finalComponents} remarkPlugins={remarkPlugins} rehypePlugins={rehypePlugins}>
            {children}
          </ActualReactMarkdown>
        </div>
      );
    },
  };
});

// Mock SyntaxHighlighter
jest.mock('react-syntax-highlighter', () => ({
  Prism: {
    SyntaxHighlighter: ({ children, language, ...props }: any) => (
      <pre data-testid="mock-syntax-highlighter" className={`language-${language}`} {...props}>
        <code data-testid="mock-syntax-highlighter-code">{children}</code>
      </pre>
    ),
  },
  nightOwl: {},
  prism: {},
}));


const renderWithProviders = (content: string, selectedFilePath: string | null = null) => {
  const theme = createTheme({
    palette: {
      mode: 'light',
    },
  });
  return render(
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <MarkdownRenderer content={content} selectedFilePath={selectedFilePath} />
      </ThemeProvider>
    </BrowserRouter>
  );
};

describe('MarkdownRenderer', () => {
  const mockNavigate = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (useNavigate as jest.Mock).mockReturnValue(mockNavigate);
  });

  it('renders markdown content correctly', () => {
    const content = '# Hello World\n\nThis is a paragraph.';
    const { asFragment } = renderWithProviders(content);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByRole('heading', { name: 'Hello World' })).toBeInTheDocument();
    expect(screen.getByText('This is a paragraph.')).toBeInTheDocument();
  });

  it('renders code blocks with syntax highlighting', () => {
    const content = '```javascript\nconsole.log(\"Hello\");\n```';
    renderWithProviders(content);
    expect(screen.getByTestId('mock-syntax-highlighter')).toBeInTheDocument();
    expect(screen.getByTestId('mock-syntax-highlighter-code')).toHaveTextContent('console.log(\"Hello\");');
    expect(screen.getByTestId('mock-syntax-highlighter')).toHaveClass('language-javascript');
  });

  it('renders mermaid diagrams for mermaid code blocks', () => {
    const content = '```mermaid\ngraph TD; A-->B;\n```';
    renderWithProviders(content);
    expect(screen.getByTestId('mock-mermaid-chart')).toBeInTheDocument();
    expect(screen.getByTestId('mock-mermaid-chart')).toHaveTextContent('graph TD; A-->B;');
  });

  it('resolves relative paths correctly for internal links', () => {
    const content = '[Link to another file](another.md)';
    renderWithProviders(content, '/path/to/current.md');
    const link = screen.getByTestId('markdown-link');
    expect(link).toHaveAttribute('href', '/path/to/another.md');
  });

  it('navigates to the correct path on internal link click', () => {
    const content = '[Link to another file](another.md)';
    renderWithProviders(content, '/path/to/current.md');
    const link = screen.getByTestId('markdown-link');
    fireEvent.click(link);
    expect(mockNavigate).toHaveBeenCalledWith('/path/to/another.md');
  });

  it('opens external links in a new tab', () => {
    const content = '[External Link](https://example.com)';
    renderWithProviders(content, '/path/to/current.md');
    const link = screen.getByTestId('markdown-link');
    expect(link).toHaveAttribute('href', 'https://example.com');
    expect(link).toHaveAttribute('target', '_blank');
    expect(link).toHaveAttribute('rel', 'noopener noreferrer');
  });

  it('handles absolute paths correctly', () => {
    const content = '[Absolute Link](/absolute/path.md)';
    renderWithProviders(content, '/path/to/current.md');
    const link = screen.getByTestId('markdown-link');
    expect(link).toHaveAttribute('href', '/absolute/path.md');
  });

  it('handles root path correctly', () => {
    const content = '[Root Link](/)';
    renderWithProviders(content, '/path/to/current.md');
    const link = screen.getByTestId('markdown-link');
    expect(link).toHaveAttribute('href', '/');
  });

  it('handles hash links correctly', () => {
    const content = '[Hash Link](#section)';
    renderWithProviders(content, '/path/to/current.md');
    const link = screen.getByTestId('markdown-link');
    expect(link).toHaveAttribute('href', '#section');
  });
});
