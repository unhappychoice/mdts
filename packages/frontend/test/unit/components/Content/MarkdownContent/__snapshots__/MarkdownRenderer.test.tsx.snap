// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MarkdownRenderer renders markdown content correctly 1`] = `
<DocumentFragment>
  <div
    class="markdown-body MuiBox-root css-x7jyj4"
  >
    <div
      data-testid="mock-react-markdown-root"
    >
      <h1
        data-testid="mock-h1"
        node="[object Object]"
      >
        Hello World
      </h1>
      

      <p
        data-testid="mock-p"
        node="[object Object]"
      >
        This is a paragraph.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
